### ISSUE TITLE
Feature Request: global average pooling layer in tf.layers

### ISSUE THREAD

**Issue by ankitvgupta** [tensorflow/tensorflow#16536]
Hello,

Can we add an implementation of global_average_pooling to tf.layers or tf.contrib.layers? It can look much like the Keras implementation here: https://www.tensorflow.org/api_docs/python/tf/keras/layers/GlobalMaxPool3D
and essentially just requires calling tf.reduce_mean.

I suspect lots of people have written functions called global_pooling that just call reduce mean, and it would be nice to have a tf.layers function that just does this for consistency/readability.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: AWS Deep Learning AMI (Conda)
- **TensorFlow version (use command below)**: 1.5
- **Python version**:  3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: K80
- **Exact command to reproduce**: N/A

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by drpngx** [comment]
@martinwicke WDYT? Contributions welcome?

**Comment by martinwicke** [comment]
@fchollet This is a classic example of where having either a viable automatic LambdaLayer, or an alternative automated way of making all functions into Layers would help.

I'm not a fan of adding every TF function as a layer. As insofar as this impacts Keras workflows (incomplete layer graph), we need to find a solution to this.

**Comment by ankitvgupta** [comment]
Sorry, how would this impact Keras workflows? I believe this is a function that already exists in tf.keras as a layer that I'm hoping to add to tf.layers.

**Comment by fchollet** [comment]
@martinwicke auto-wrapping non-layer ops as layer should not be an issue (once we figure out serialization). 

But in this specific case, a global pooling layer has valuable benefits: because it includes a `data_format` argument which can have a global default, it enables you to build convnets where you can switch from NCHW to NHWC in one click.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by ankitvgupta** [comment]
@fchollet in light of your comment, should we work on a PR for this? I can probably give it a shot.

**Comment by fchollet** [comment]
@ankitvgupta no, I recommend you use `tf.keras.layers.GlobalMaxPooling3D()` instead.

**Comment by ankitvgupta** [comment]
Got it, thanks.
